packages/react-sdk/src/components/image.tsx

```diff
@@ -1,24 +1,187 @@
 import React, { forwardRef, type ElementRef, type ComponentProps } from "react";
+import { warnOnce } from "../utils/warn-once";
 
 const defaultTag = "img";
 
-type ImageProps = ComponentProps<typeof defaultTag>;
+type ImageProps = ComponentProps<typeof defaultTag> & { quality?: number };
+
+/**
+ * max(...imageSizes) must be less then min(...deviceSizes)
+ * idea is to support images with vw size less than 100vw
+ * https://github.com/vercel/next.js/blob/canary/packages/next/shared/lib/image-config.ts
+ **/
+const imageSizes = [16, 32, 48, 64, 96, 128, 256, 384];
+const deviceSizes = [640, 750, 828, 1080, 1200, 1920, 2048, 3840];
```

Comment:
What about using breakpoints user has in the project defined? We have default values and sdk/src/css/breakpoints.ts and user can change them and add new once too

Reply:
I understand that these are popular device sizes and image sizes. Vercel has no info about used media queries on the site I would assume. Another question is if their image resize service is even designed like ours, which allows to pass any width/height dynamically.

In our case we COULD use our known breakpoints to generate optimal sizes specifically designed for the right media query?

Wdyt?

Reply:
Added algo description https://github.com/webstudio-is/webstudio-designer/blob/8663e1cd5a6c8519156a3eacb6784f45b9e3b800/packages/react-sdk/src/component-utils/image.ts#L2
With whats now and possible optimisations